Add optimization page (#145)
neo4j/docs-graphql: modules/ROOT/pages/optimization.adoc

[[optimization]]
= Optimization

> This chapter contains some optimizations to improve the performance of a production system using `@neo4j/graphql`.

Reviewer: ```suggestion
This page contains optimizations to improve the performance of a production system using `@neo4j/graphql`.
```

> == Schema Optimisations

Reviewer: ```suggestion
== Schema optimizations
```

> == Schema Optimisations

Reviewer: Will there be more optimization sections on this page in the future?

Author: Yes, the plan is to add a few things that we are planning, for "Schema" and "Queries"

> This section covers improvements to reduce the size of the generated schema. Reducing the schema size has the following performance benefits:

Reviewer: ```suggestion
This section covers optimizations to reduce the size of the generated schema. Reducing the schema size has the following performance benefits:
```

> * Reduce the startup time of the server.

Reviewer: ```suggestion
* Reduce server startup time.
```

* Reduce memory footprint.

=== Exclude `@deprecated` fields
The `@neo4j/graphql` library generates some GraphQL fields and operations marked as `@deprecated`. These exists to keep compatibility with previous versions of the library.

> If you are not using these deprecated fields, you can disable generation for these with the `excludeDeprecatedFields` flag in the library setup. 

Reviewer: ```suggestion
If you are not using these deprecated fields, you can disable their generation with the `excludeDeprecatedFields` flag in the library setup. 
```
